--- origsrc/gourmet-0.13.4/setup.py	2006-05-03 19:54:27.000000000 -0500
+++ src/gourmet-0.13.4/setup.py	2007-06-03 00:41:54.359375000 -0500
@@ -47,7 +47,7 @@
         raise
     mod_list = [#'metakit',
         'Image',
-        'pysqlite2',
+        'sqlite3',
         ]
     ok = 1
     for m in mod_list:
--- origsrc/gourmet-0.13.4/src/lib/backends/sqlite_db.py	2006-12-29 12:54:38.000000000 -0600
+++ src/gourmet-0.13.4/src/lib/backends/sqlite_db.py	2007-06-03 00:42:41.906250000 -0500
@@ -1,5 +1,5 @@
 import sql_db, rdatabase
-from pysqlite2 import dbapi2 as sqlite
+from sqlite3 import dbapi2 as sqlite
 import os, os.path, re
 import gourmet.gglobals as gglobals
 from gourmet import keymanager
